### PR TITLE
WIZZIE-3406 Create Prozzie AMI packer spec, provisioning and bootstrapping scripts

### PR DESCRIPTION
**Pull request by cjmateos**

### CODE REVIEW

File: cookbooks/bootstrap/prozziebootstrap/libraries/helper.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module ProzzieBootstrap
+  module Helper
+
+    def public_ip
+      require 'net/http'
+      require 'uri'
+
+      uri_ip = URI.parse("http://169.254.169.254/2018-03-28/meta-data/public-ipv4")
```
Comment (andresgomezfrr) [inline cookbooks/bootstrap/prozziebootstrap/libraries/helper.rb]: What is the meaning of this date `2018-03-28`?

File: cookbooks/bootstrap/prozziebootstrap/metadata.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name             'prozziebootstrap'
+maintainer       'Wizzie Analytics'
+maintainer_email 'devel@wizzie.io'
+license          'All rights reserved'
+description      'Bootstraps Prozzie Marketplace AMI'
+version          '0.0.1'
```
Comment (andresgomezfrr) [inline cookbooks/bootstrap/prozziebootstrap/metadata.rb]: Maybe we need to use something like WDP AMI. ${PROZZIE_VERSION}_1

File: cookbooks/bootstrap/prozziebootstrap/resources/config.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+action :config do
+
+  prozzie_dir = '/usr/local/etc/prozzie/ami'
+  wizzie_home = '/home/wizzie'
+  user_data_file = '/var/lib/cloud/instance/user-data.txt'
+
+  user_data = {}
+  begin
+    user_data = YAML.load_file(user_data_file)
+    user_data = {} unless user_data
```
Comment (andresgomezfrr) [inline cookbooks/bootstrap/prozziebootstrap/resources/config.rb]: I think that this line isn't necessary. `user_data` is initialized before the begin.

File: cookbooks/provision/prozzieprov/metadata.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+name             'prozzieprov'
+maintainer       'Wizzie Analytics'
+maintainer_email 'devel@wizzie.io'
+license          'All rights reserved'
+description      'Provision Prozzie Marketplace AMI'
+version          '0.0.1'
```
Comment (andresgomezfrr) [inline cookbooks/provision/prozzieprov/metadata.rb]: Maybe we need to use something like WDP AMI. ${PROZZIE_VERSION}_1

File: cookbooks/provision/prozzieprov/resources/config.rb [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  [ 'docker.io' ].each do |pack|
+    package pack do
+      action :install
+    end
+  end
+
+  bash 'install docker-compose' do
+    user 'root'
+    code <<-EOH
+    curl -s -L "https://github.com/docker/compose/releases/download/1.21.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/bin/docker-compose
```
Comment (andresgomezfrr) [inline cookbooks/provision/prozzieprov/resources/config.rb]: You are fixing the docker-compose version. Is it ok?

File: cookbooks/bootstrap/prozziebootstrap/libraries/helper.rb (hunk 1 above)
Comment (cjmateos) [reply cookbooks/bootstrap/prozziebootstrap/libraries/helper.rb]: It is a specific endpoint version of the internal metadata API.

```
$ curl http://169.254.169.254/
1.0
2007-01-19
2007-03-01
2007-08-29
2007-10-10
2007-12-15
2008-02-01
2008-09-01
2009-04-04
2011-01-01
2011-05-01
2012-01-12
2014-02-25
2014-11-05
2015-10-20
2016-04-19
2016-06-30
2016-09-02
2018-03-28
2018-08-17
2018-09-24
latest
```

File: cookbooks/bootstrap/prozziebootstrap/metadata.rb (hunk 2 above)
Comment (cjmateos) [reply cookbooks/bootstrap/prozziebootstrap/metadata.rb]: That version number is related with the `prozziebootstrap` cookbook, not about the GitHub project or tha AMI version. The AMIM version will be tagged in the GitHub project

File: cookbooks/provision/prozzieprov/metadata.rb (hunk 4 above)
Comment (cjmateos) [reply cookbooks/provision/prozzieprov/metadata.rb]: That version number is related with the `prozzieprov` cookbook, not about the GitHub project or tha AMI version. The AMIM version will be tagged in the GitHub project

File: cookbooks/bootstrap/prozziebootstrap/resources/config.rb (hunk 3 above)
Comment (cjmateos) [reply cookbooks/bootstrap/prozziebootstrap/resources/config.rb]: This line is to protect the cookbook when there is not user_data. In that case (empty user_data), if we try to get some value of the YAML file (for example, when `template "#{prozzie_dir}/prozzie.conf"` is loaded), an error will happen.

File: cookbooks/provision/prozzieprov/resources/config.rb (hunk 5 above)
Comment (cjmateos) [reply cookbooks/provision/prozzieprov/resources/config.rb]: That's it

**Pull request merged**